OnlinerTask/OnlinerTask.BLL/Services/Products/Implementations/Manager.cs

```diff
@@ -25,14 +28,16 @@ protected Manager(ISearchService searchService, IRepository repository)
             {
                 return;
             }
-            repository.CreateOnlinerProduct(result, name);
+            var productId = repository.CreateOnlinerProduct(result, name);
             AddNotify(result.FullName);
+            await productLogger.LogAdding(productId, result);
```

Comment:
Logging should be deferred as much as possible. Try to rethink how logs should be stored.